File: run/Core/Manila.js
const project = Manila.getProject()
const workspace = Manila.getWorkspace()

Manila.apply('shiron.manila:cpp@1.0.0:staticlib')
const config = Manila.getConfig()

project.Version('1.0.0')
project.Description('Demo Project Core')

project.SourceSets({
	main: Manila.sourceSet(project.GetPath().join('src/main')).include('**/*.cpp'),
	test: Manila.sourceSet(project.GetPath().join('src/test')).include('**/*.cpp')
})

project.Artifacts({
	main: Manila.artifact(artifact => {
		Manila.job('build')
			.description('Build the Core')
			.execute(async () => {
				await Manila.build(workspace, project, config, artifact)
			})
	})
		.from('shiron.manila:cpp/console')
		.description('Core Main Artifact')
})

File: run/Zip/Manila.cs
var project = Manila.GetProject();
var workspace = Manila.GetWorkspace();

project.Version("1.0.0");
project.Description("Demo Project Core");

project.SourceSets(new Dictionary<string, object> {
    ["main"] = Manila.SourceSet(project.GetPath().Join("main")).Include("**/*")
});

project.Artifacts(new Dictionary<string, object> {
    ["main"] = Manila.Artifact("shiron.manila:zip/zip", artifact => {
        var config = Manila.GetConfig<ZipBuildConfig>(artifact);
        config.SetSubFolder(Manila.GetEnv("MANILA_SUB_FOLDER", "sub"));

        Manila.Job("build")
            .Description("Create the Zip File")
            .Execute(async () => {
                await Manila.Build(project, config, artifact);
            });
    }).Description("Zip Main Artifact")
});
